src/viem/estimateFeesPerGas.test.ts

```diff
@@ -25,5 +38,30 @@ describe(estimateFeesPerGas, () => {
     }
     const fees = await estimateFeesPerGas(client as any, '0x123')
     expect(fees).toEqual({ maxFeePerGas: BigInt(110), maxPriorityFeePerGas: BigInt(10) })
+    expect(defaultEstimateFeesPerGas).not.toHaveBeenCalled()
+  })
+
+  it('should return the default fees per gas on other networks', async () => {
+    jest
+      .mocked(defaultEstimateFeesPerGas)
+      .mockResolvedValue({ maxFeePerGas: BigInt(110), maxPriorityFeePerGas: BigInt(10) })
+    const client = {
+      chain: { id: 1 },
+    }
+    const fees = await estimateFeesPerGas(client as any)
+    expect(fees).toEqual({ maxFeePerGas: BigInt(110), maxPriorityFeePerGas: BigInt(10) })
+    expect(defaultEstimateFeesPerGas).toHaveBeenCalledWith(client)
+  })
+
+  it('should return the default fees per gas on other networks when fee currency is specified', async () => {
+    jest
+      .mocked(defaultEstimateFeesPerGas)
+      .mockResolvedValue({ maxFeePerGas: BigInt(110), maxPriorityFeePerGas: BigInt(10) })
+    const client = {
+      chain: { id: 1 },
+    }
+    const fees = await estimateFeesPerGas(client as any, '0x123')
```

Comment:
nit: As far as I understand the code, passing the `feeCurrency` here is pointless.
I wonder if this might be confusing, for example, I want to use some specific token as fee currency, but passing it here won't do the job.

I'm not sure how is the best way to solve this, since currently only Celo supports this feature.
I'm wondering if we should just throw if `feeCurrency` is not undefined for non-Celo transactions.
I think the behavior should be similar to what happens when we pass a invalid `feeCurrency` for Celo. Does `eth_gasPrice` fail when the `feeCurrency` is invalid?

Reply:
Yes good idea to throw if not Celo.